idioms/mem-replace.md

```diff
@@ -36,6 +36,28 @@ fn a_to_b(e: &mut MyEnum) {
 }
 ```
 
+This also works with more variants:
+
+```Rust
+use std::mem;
+
+enum MultiVariateEnum {
+    A { name: String },
+    B { name: String },
+    C,
+    D
+}
+
+fn swizzle(e: &mut MultiVariateEnum) {
```

Comment:
Removing the `mem::replace(..)` requires taking `e` by value, otherwise borrowck will not allow you to take out `name` because it's just borrowed, not owned. One could of course take `e` by value, but there may be reasons against that (e.g. API ergonomics).